File: part-2/module-5-multilingual-design/workspace_folder/src/example_pkg_py/example_pkg_py/testsub.py
#Daniel Yanke
#10/3/2024

import rclpy
from rclpy.node import Node

from example_test_interfaces.msg import ExampleTest

class PySubscriber(Node):
    def __init__(self):
        super().__init__('py_subscriber')
        self.subscriber = self.create_subscription(
            ExampleTest,
            'test_topic',
            self.subscription_callback,
            10
        )
    def subscription_callback(self, msg):
        self.get_logger().info(f"I heard: '{msg.unity_id}', '{msg.seq_number}' from {msg.language} node")

def main(args = None):
    rclpy.init(args = args)
    python_subscriber = PySubscriber()
    rclpy.spin(python_subscriber)
    python_subscriber.destroy_node()
    rclpy.shutdown()

if __name__ == "__main__":
    main()

File: part-2/module-5-multilingual-design/workspace_folder/src/example_pkg_cpp/src/testsub.cpp
/*
Author: Daniel Yanke
Date: 10/3/2024
*/

#include <memory>
#include <string>

#include "rclcpp/rclcpp.hpp"
#include "example_test_interfaces/msg/example_test.hpp"
using std::placeholders::_1;

class CppSubscriber : public rclcpp::Node{
public:
    CppSubscriber() : Node("cpp_subscriber"){
        subscription_ = this->create_subscription<example_test_interfaces::msg::ExampleTest>(
            "test_topic", 10, std::bind(&CppSubscriber::topic_callback, this, _1)
        );
    }
private:
    void topic_callback(const example_test_interfaces::msg::ExampleTest & msg) const{
        RCLCPP_INFO(this->get_logger(), "I heard: '%s', '%s' from %s node", msg.unity_id.c_str(), std::to_string(msg.seq_number).c_str(), msg.language.c_str());
    }
    rclcpp::Subscription<example_test_interfaces::msg::ExampleTest>::SharedPtr subscription_;
};

int main(int argc, char * argv[])
{
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<CppSubscriber>());
    rclcpp::shutdown();
    return 0;
}